### PR TITLE
removes vault health service

### PR DESCRIPTION
**Pull request by xh3b4sd**
Fixes https://github.com/giantswarm/giantswarm/issues/3547.

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -22,12 +22,12 @@
 
 
 [[constraint]]
-  name = "github.com/cenkalti/backoff"
```
Comment (xh3b4sd) [inline Gopkg.toml]: Vendoring our new backoff.

File: main.go [hunk 2]
```diff
@@ -47,7 +47,7 @@ func main() {
 
 			serviceConfig.Description = description
 			serviceConfig.GitCommit = gitCommit
-			serviceConfig.Name = name
+			serviceConfig.ProjectName = name
```
Comment (xh3b4sd) [inline main.go]: Boyscouting.

File: server/endpoint/endpoint.go [hunk 3]
```diff
@@ -13,21 +13,15 @@ import (
...
-	return Config{
-		// Dependencies.
-		Logger:     nil,
-		Middleware: nil,
-		Service:    nil,
-	}
+// Endpoint is the endpoint collection.
+type Endpoint struct {
+	Healthz *healthz.Endpoint
+	Version *version.Endpoint
```
Comment (xh3b4sd) [inline server/endpoint/endpoint.go]: Boyscouting.

File: server/endpoint/endpoint.go [hunk 4]
```diff
@@ -36,24 +30,27 @@ func New(config Config) (*Endpoint, error) {
...
 	{
-		healthzConfig := healthz.DefaultConfig()
-		healthzConfig.Logger = config.Logger
-		healthzConfig.Services = []healthzservice.Service{
-			config.Service.Healthz.K8s,
-			config.Service.Healthz.Vault,
+		c := healthz.Config{
+			Logger: config.Logger,
+			Services: []healthzservice.Service{
+				config.Service.Healthz.K8s,
```
Comment (tuommaki) [inline server/endpoint/endpoint.go]: This is still checking against k8s API. Weren't we supposed to drop those as well?

Comment (xh3b4sd) [reply server/endpoint/endpoint.go]: Uncertain. The postmortem and the issue were only about Vault dependency. We did not have huge problems with the k8s healthcheck so far. We should maybe think about the health checking practises in general and derive some guidelines we can apply so we get it straight once and for all.

Comment (tuommaki) [reply server/endpoint/endpoint.go]: Roger that.

**Pull request merged**